aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -571,95 +571,105 @@ Tensor sparse_compressed_to_dense(
       !dtype.has_value(),
       "dtype argument is not supported by sparse_csr_to_dense");
 
-  // Guard upfront against hybrid tensors (causes segfault)
-  auto batch_ndim = sparse_csr::numBatchDimensions(self);
+  if (self.layout() == kSparseCsr || self.layout() == kSparseCsc
+      || self.layout() == kSparseBsr || self.layout() == kSparseBsc) {
```

Comment:
`self` in `sparse_compressed_to_dense` has always one of these layouts, so:
```suggestion
```
(also remove the closing curly bracket of the if-statement).